Backup important wiki information

carlossanlop opened the pull request:
Fixes: https://github.com/dotnet/corefx/issues/36057

Original documents in this PR:
- ~~[Building .NET Core 2.X on FreeBSD](https://github.com/dotnet/corefx/wiki/Building-.NET-Core--2.x-on-FreeBSD)~~
- [Building .NET Core 3.X on FreeBSD](https://github.com/dotnet/corefx/wiki/Building-.NET-Core-3.x-on-FreeBSD)
- ~~[Links](https://github.com/dotnet/corefx/wiki/Links)~~
- [Pick Issue](https://github.com/dotnet/corefx/wiki/Pick-issue) - Only rescued two labels mentioned here that we were not mentioning in the CoreFX documentation.
- Also added the links to the FreeBSD instructions in unix-instructions.md.

carlossanlop committed:
Backup important wiki information

carlossanlop committed:
Restore original title in unix-instructions.md

jkotas commented on Documentation/building/freebsd-instructions-2.x.md:
```diff
@@ -0,0 +1,390 @@
+# FreeBSD 3.X build instructions
```
The file name says 2.x, but the title says 3.x.

This should just say FreeBSD build instructions, and we should have just the latest ones. We version the build instructions together with the code. E.g. master has build instructions for master; release/2.0 has the build instructions for release/2.0; etc.

carlossanlop replied:
These are the instructions to build 2.X. I had a typo. But based on your comment, I'll remove the file.

carlossanlop committed:
FreeBSD instructions only for 3.0

jkotas commented on Documentation/project-docs/conferences.md:
```diff
@@ -0,0 +1,47 @@
+## .NET Conferences and Announcements links
```
I do not see a single conference listed in this list

jkotas commented on Documentation/project-docs/conferences.md:
```diff
@@ -0,0 +1,47 @@
+    * [.NET Core 2.0 perf improvements](https://blogs.msdn.microsoft.com/dotnet/2017/06/07/performance-improvements-in-net-core/)
+* [.NET Standard versions](https://github.com/dotnet/standard/blob/master/docs/versions.md)
+    * Next .NET Standard plans - [.NET Standard 2.1](https://github.com/dotnet/standard/blob/master/docs/planning/netstandard-2.1/README.md#net-standard-21)
+* [CoreFX contributors](https://github.com/dotnet/corefx/graphs/contributors)
```
Why is this here? It is neither conference or announcement.

jkotas commented on Documentation/project-docs/conferences.md:
```diff
@@ -0,0 +1,47 @@
+### .NET plans
+
+* .NET Core 3.1 - LTS - November 2019 (see [.NET 5 announcement blog](https://devblogs.microsoft.com/dotnet/introducing-net-5/))
+* [.NET 5](https://devblogs.microsoft.com/dotnet/introducing-net-5/) - November 2020 GA
```
This is redundant with roadmap on https://github.com/dotnet/core. I do not think we need to be maintaining a duplicate information here.

danmosemsft replied:
I agree, I think if we need such a document as this it should probably be in dotnet/core. I'd dump it.

jkotas commented on Documentation/project-docs/conferences.md:
```diff
@@ -0,0 +1,47 @@
+
+### .NET Core 3.0
+
+* [.NET Core 3.0 previews](https://github.com/dotnet/core/tree/master/release-notes/3.0)
```
Information about 3.0 previews is irrelevant now that 3.0 shipped. 

This whole page looks very questionable to me. Is it really useful?

stephentoub replied:
Information on this page is the purpose of the dotnet/core repo.  I would just delete this whole .md file.

danmosemsft commented on Documentation/building/freebsd-instructions.md:
```diff
@@ -0,0 +1,278 @@
+# FreeBSD build instructions
+
+## Disclaimer
+Instructions bellow may or may not give you what you want.
```
typo

stephentoub replied:
In addition to the "bellow" typo, this is also a strange disclaimer.  Why have instructions if they "may or may not give you what you want?"  We should be clear about why they may not provide the desired results.

stephentoub commented on Documentation/building/freebsd-instructions.md:
````diff
@@ -0,0 +1,278 @@
+
+```sudo pkg install cmake git icu libunwind bash python2 krb5 lttng-ust llvm60 libgit2```
+
+some scripts may still assume /bin/bash exists. To workaround it for now do of needed:
````
"now do of needed"?

stephentoub commented on Documentation/building/freebsd-instructions.md:
````diff
@@ -0,0 +1,278 @@
+sudo ln -s /usr/local/bin/bash /bin/bash
+```
+
+This is certainly undesirable and it should be avoided if possible.
````
What is undesirable?

stephentoub commented on Documentation/building/freebsd-instructions.md:
````diff
@@ -0,0 +1,278 @@
+cd ~/dotnet
+curl https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-sdk-latest-freebsd-x64.tar.gz | tar xfz - 
+```
+if on 12.x you may also need to set `LD_PRELOAD` to `/usr/lib/libpthread.so` to avoid issue when cli freezes. 
````
"to avoid issue" => "to avoid issues"

And what does it mean that "cli freezes"?

stephentoub commented on Documentation/building/freebsd-instructions.md:
```diff
@@ -0,0 +1,278 @@
+if on 12.x you may also need to set `LD_PRELOAD` to `/usr/lib/libpthread.so` to avoid issue when cli freezes. 
+
+
+As of summer 2019 this CLI is no longer good enough to build all repos. If that is your case jump to section [Updating CLI](#updating--bootstrap-cli)
```
"If that is your case"... what case?  It's beyond that point for everyone, no?

stephentoub commented on Documentation/building/freebsd-instructions.md:
```diff
@@ -0,0 +1,278 @@
+Binary snapshot can be obtained from https://github.com/wfurt/blob as dotnet-sdk-freebsd-x64-latest.tgz
+
+## Getting sources
+master of source-build pulls in source code of specific snapshot instead of tip of master branches. 
```
I'm having a hard time following these docs.  I'm going to stop commenting on it.

carlossanlop replied:
Yes I see what you mean. The document needs to be rewritten.

stephentoub commented on Documentation/building/unix-instructions.md:
```diff
@@ -1,5 +1,8 @@
 Building CoreFX on FreeBSD, Linux and OS X
 ==========================================
+
+Build instructions for FreeBSD can be found [here](freebsd-instructions.md)
```
FreeBSD isn't supported, whereas Linux and macOS are.  Seems like "Building CoreFX on FreeBSD, Linux, and OSX" should instead be "Building CoreFX on Linux and macOS", and then this line should be in its own section later.

carlossanlop replied:
I'll move it to its own section at the end of this document.

carlossanlop committed:
Remove conferences.md

carlossanlop committed:
update unix-instructions.md title

carlossanlop commented:
Thanks for your comments, @stephentoub @jkotas @danmosemsft .

- I removed the conferences file. It seems it only contains links to announcements which, as Stephen mentioned, we already post regularly on dotnet/core.
- I removed "FreeBSD" from the title in `unix-instructions.md`.
- I still think we should bring the FreeBSD 3.X instructions to corefx, since the document has been modified recently multiple times (see the [commit history](https://github.com/dotnet/corefx/wiki/Building-.NET-Core-3.x-on-FreeBSD/_history)). The main problem is the language used to write the document, so we could ask the people involved in this document in the past to help reword the document, or I can do it myself, but I would do it in a future PR. What are your thoughts?

stephentoub commented:
Sounds fine. Thanks.

carlossanlop committed:
Move FreeBSD link to the end of unix-instructions.md
